Fails to build on macos 10.15:

```
Undefined symbols for architecture x86_64:
  "_BIO_ctrl", referenced from:
      _dynsec_auth__base64_encode in auth.c.o
      _dynsec_auth__base64_decode in auth.c.o
  "_BIO_f_base64", referenced from:
      _dynsec_auth__base64_encode in auth.c.o
      _dynsec_auth__base64_decode in auth.c.o
  "_BIO_free_all", referenced from:
      _dynsec_auth__base64_encode in auth.c.o
      _dynsec_auth__base64_decode in auth.c.o
  "_BIO_new", referenced from:
      _dynsec_auth__base64_encode in auth.c.o
      _dynsec_auth__base64_decode in auth.c.o
  "_BIO_push", referenced from:
      _dynsec_auth__base64_encode in auth.c.o
      _dynsec_auth__base64_decode in auth.c.o
  "_BIO_read", referenced from:
      _dynsec_auth__base64_decode in auth.c.o
  "_BIO_s_mem", referenced from:
      _dynsec_auth__base64_encode in auth.c.o
      _dynsec_auth__base64_decode in auth.c.o
  "_BIO_set_flags", referenced from:
      _dynsec_auth__base64_encode in auth.c.o
      _dynsec_auth__base64_decode in auth.c.o
  "_BIO_write", referenced from:
      _dynsec_auth__base64_encode in auth.c.o
      _dynsec_auth__base64_decode in auth.c.o
  "_EVP_get_digestbyname", referenced from:
      _dynsec_auth__pw_hash in auth.c.o
      _dynsec_auth__basic_auth_callback in auth.c.o
  "_PKCS5_PBKDF2_HMAC", referenced from:
      _dynsec_auth__pw_hash in auth.c.o
      _dynsec_auth__basic_auth_callback in auth.c.o
  "_RAND_bytes", referenced from:
      _dynsec_auth__pw_hash in auth.c.o
  "_mosquitto_broker_publish", referenced from:
      _dynsec_control_callback in plugin.c.o
  "_mosquitto_callback_register", referenced from:
      _mosquitto_plugin_init in plugin.c.o
  "_mosquitto_callback_unregister", referenced from:
      _mosquitto_plugin_cleanup in plugin.c.o
  "_mosquitto_calloc", referenced from:
      _dynsec_auth__base64_decode in auth.c.o
      _dynsec_clients__config_load in clients.c.o
      _dynsec_clients__process_create in clients.c.o
      _dynsec_groups__config_load in groups.c.o
      _dynsec_groups__process_create in groups.c.o
      _mosquitto_plugin_init in plugin.c.o
      _dynsec_roles__config_load in roles.c.o

<snip>

ld: symbol(s) not found for architecture x86_64
```

--- a/pkgs/servers/mqtt/mosquitto/default.nix
+++ b/pkgs/servers/mqtt/mosquitto/default.nix
@@ -12,18 +12,28 @@
 , openssl
 , withSystemd ? stdenv.isLinux
 , systemd
+, fetchpatch
 }:
 
 stdenv.mkDerivation rec {
   pname = "mosquitto";
-  version = "2.0.10";
+  version = "2.0.11";
 
   src = fetchFromGitHub {
     owner = "eclipse";
     repo = pname;
     rev = "v${version}";
-    sha256 = "144vw7b9ja4lci4mplbxs048x9aixd9c3s7rg6wc1k31w099rb12";
+    sha256 = "1i0kkqhjxqahkhsl60s46zlywa87jakv7h4sr3fhi1v53vwpb9mg";
   };
+
+  patches = lib.optionals stdenv.isDarwin [
+    (fetchpatch {
+      name = "revert-cmake-shared-to-module.patch"; # See https://github.com/eclipse/mosquitto/issues/2277
+      url = "https://github.com/eclipse/mosquitto/commit/e21eaeca37196439b3e89bb8fd2eb1903ef94845.patch";
+      sha256 = "14syi2c1rks8sl2aw09my276w45yq1iasvzkqcrqwy4drdqrf069";
+      revert = true;
+    })
+  ];
 
   postPatch = ''
     for f in html manpage ; do

